qtt/algorithms/fitting.py

```diff
@@ -53,10 +53,24 @@ def initFermiLinear(x_data, y_data, fig=None):
         # subtract linear part
         yr = ydata - y
 
-        cc = np.mean(xdata)
+        sigma = xdata.size/250
+        dyr = scipy.ndimage.gaussian_filter(yr, sigma, order=1)
+        
+        # assume step is steeper than overall slope
+        cc_idx = np.argmax(np.abs(dyr))            
+        h = int(xdata.size / 2)
+        
+        # prevent guess to be at the edges
+        if cc_idx < xdata.size/100 or cc_idx > (99/100)*xdata.size:
+            cc = np.mean(xdata)
+            A = -(np.mean(yr[h:]) - np.mean(yr[:h]))
```

Comment:
Why not
```suggestion
             A = -1*np.sign(dyr[cc_idx])*(np.mean(yr[h:]) - np.mean(yr[:h]))
```
As for the other case?